Maxima_Minima.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Sep  5 16:25:26 2020
+
+@author: INPRDHI
+"""
+import os.path  as path
+
+def strictly_increasing(arr):
+    return all(x<y for x, y in zip(arr, arr[1:]))
+
+def strictly_decreasing(arr):
+    return all(x>y for x, y in zip(arr, arr[1:]))
+
+# Python program of above implementation 
+def getMinMax(low, high, arr):    
+    
+    arr_max = arr[low] 
+    arr_min = arr[low] 
+    
+    # print(arr_max, arr_min,arr)
+      
+    # If there is only one element  
+    # if low == high: 
+    #     print("when equal")
+    #     arr_max = arr[low] 
+    #     arr_min = arr[low] 
+    #     return (arr_min,arr_max) 
+          
+    strict_increase = strictly_increasing(arr)
+    # print(strict_increase)
+    strict_decrease = strictly_decreasing(arr)
+    
+    # If there is only two element 
+    if high == low + 1: 
+        if (strict_decrease): 
+            arr_max = arr[low] 
+            arr_min = arr[high] 
+            # print("dec")
+            return ("decreasing", arr_min,arr_max)
+        if (strict_increase): 
+            arr_max = arr[high] 
+            arr_min = arr[low] 
+            # print("inc")
+            return ("increasing", arr_min,arr_max) 
+        else:
+            if(arr[low] < arr[high]):
+                arr_max = arr[high] 
+                arr_min = arr[low] 
+                return ("maxima", arr_min,arr_max) 
+            else:
+                 arr_max = arr[low] 
+                 arr_min = arr[high]
+                 return ("minima", arr_min,arr_max)  
+                
+    else: 
+          
+        # If there are more than 2 elements 
+        mid = int((low + high) / 2) 
+        maxMin1, arr_min1, arr_max1 = getMinMax(low, mid, arr) 
+        
+        maxMin2, arr_min2, arr_max2 = getMinMax(mid + 1, high, arr) 
+       
+    # print(maxMin1, ":",arr_min1,":",arr_max1)
+    # print(maxMin2, ":",arr_min2,":",arr_max2)
+    
+    if(maxMin1 == maxMin2):
+         return (maxMin1, min(arr_min1, arr_min2),max(arr_max1, arr_max2) )
+     
+    elif(maxMin1 == "decreasing" or maxMin1 == "minima"):
+        return("minima", min(arr_min1, arr_min2), max(arr_max1, arr_max2))
+    else: 
+        return("maxima", min(arr_min1, arr_min2), max(arr_max1, arr_max2))
+      
+  
+   # return (max(arr_max1, arr_max2), min(arr_min1, arr_min2)) 
+  
+# Driver code 

```

Comment:
move driver code into a separate class. Preferably as an UT. You can explore pytest unit test framework for this.

This will also help you to identify the coverage of your code.

Reply:
not needed but would like to learn how it works